hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationServiceSupport.java

```diff
@@ -176,6 +178,9 @@ public void cleanResources(ConstructorFunction<Object, Throwable> responseCtor,
                 invocation.notifyException(responseCtor.createNew(null));
             }
         }
+        //Try cleaning again later to compensate the Race between `send` and `cleanResources`
```

Comment:
This task is scheduled on internal executor which does not make any blocking call. Although it should not be problem,  invocation monitor approach seems cleaner to me, mostly because I don't want to choose a big number here to prevent race and make the user wait longer.

About scanning invocations every second, the data structure that we are using is ConcurrentHashMap here.
This hashmap keeps callId -> invocation and accessed in each invocation and invocation response. 
Do we think that, iterating through this to find dead connections also cause a significant drop in performance.